### ISSUE TITLE
Switching to custom ammo types resulting in no ammo

### ISSUE THREAD

**Issue by TheOnly8Z** [Z0mbine/GAM#2]
How to replicate issue:
1. Enter singleplayer. I have almost no addons installed besides GAM, and they do not conflict.
2. Spawn a weapon(I used 357) and hold it. (I have sbox_weapons 0 on)
3. Use any ammo (I used plasma), and then switch to it using cycleweapon.
4. Now even if I had used plasma ammo, the total ammo amount is 0. Currently loaded ammo is also 0.
5. Spawning weapon-specific (for me, 357 ammo) loads GAM ammo as usual, and ammo types are usable. After shooting some ammo ( I fired about 25 rounds with 1 plasma ammo) it reverts to default.
6. Switching back to default ammo will have previously loaded ammo return to reserve.

This is a pretty major issue, please look into it. Thanks :8ball:

**Comment by zambony** [comment]
Yeah, I am aware. The reason being is I can't tell the game to change what kind of ammo a base-game weapon uses, as they aren't created using lua. There is one way around it, but it results in the client being unable to see their ammo count, and causes minor networking issues.
My intent with this addon is that it be used with SWEPs, quite honestly, but if enough people want HL2 weapon support, I'll try to add it.
Sorry.

**Comment by TheOnly8Z** [comment]
So default hl2 weapons won't work, but other SWEPs will?

**Comment by TheOnly8Z** [comment]
What about a custom HUD that shows custom ammo left?

**Comment by zambony** [comment]
Right, most, if not all, SWEPs should be able to accept the custom ammo.

I had thought about a HUD, but at the same time, I can't account for where other HUDs place their information, and SWEP bases like FAS and CW2 already make floating text on the weapon's barrel, so there's really nowhere "safe" to put that kind of info.
